packages/firestore/src/util/async_queue.ts

```diff
@@ -27,12 +27,17 @@ type TimerHandle = any;
  * Wellknown "timer" IDs used when scheduling delayed operations on the
  * AsyncQueue. These IDs can then be used from tests to check for the presence
  * of operations or to run them early.
+ *
+ * The string values are used when encoding these timer IDs in JSON spec tests.
  */
 export enum TimerId {
-  ListenStreamIdle,
-  ListenStreamConnection,
-  WriteStreamIdle,
-  WriteStreamConnection
+  // All can be used with runDelayedOperationsEarly() to run all timers.
+  All = 'all',
+  ListenStreamIdle = 'listen_stream_idle',
+  ListenStreamConnection = 'listen_stream_connection',
+  WriteStreamIdle = 'write_stream_idle',
+  WriteStreamConnection = 'write_stream_connection',
+  OnlineStateTimeout = 'online_state_timeout'
```

Comment:
I don't understand how OnlineStateTimeout is different from either the connection or idle timeouts. I'm not sure if I'm asking for it to be deduplicated against those others, comments on the individual timer ids, or some kind of theory of operation primer on how these all fit together.

Reply:
I added comments that hopefully makes things somewhat clearer.  FWIW, coming up with names for these is a bit tricky because they're meant to be high-level concepts (since they'll be triggered from high-level tests, etc.) and so e.g. "Connection" was chosen over "Backoff"